lib/msf/core/modules/metadata/cache.rb

```diff
@@ -91,15 +106,41 @@ def get_unchanged_module_references
   private
   #######
 
+  def remove_from_cache(module_name)
+    @module_metadata_cache.each do |cache_key, module_metadata|
+
+      if module_metadata.ref_name.eql? module_name
+        @module_metadata_cache.delete cache_key
```

Comment:
Could this method be more cleanly written without the loop by using the `delete_if` method?
```
orig_size = @module_metadata_cache.size
@module_metadata_cache.delete_if {|cache_key, module_metadata| module_metadata.ref_name.eql? module_name }
return orig_size != @module_metadata_cache.size
```

Reply:
sure